client/web/src/user/profile/UserProfile.tsx

```diff
@@ -40,7 +61,17 @@ export const UserProfile: FC<Pick<UserAreaRouteContext, 'user'>> = ({ user }) =>
                 data.visible ? (
                     <Fragment key={data.name}>
                         <dt>{data.name}</dt>
-                        <dd>{data.value}</dd>
+                        <dd>
+                            {data.isList ? (
+                                <ul>
+                                    {data.value.map((value, index) => (
+                                        <li key={index}>{value}</li>
```

Comment:
Using the index as the key can lead to weird re-rendering bugs if the order of roles ever changes while you're on this page. Since we know role names should be unique, too, I think we can use them as the key:

```suggestion
                                        <li key={value}>{value}</li>
```